packages/node_modules/@ciscospark/internal-plugin-ediscovery/src/ediscovery.js

```diff
@@ -0,0 +1,520 @@
+   * @returns {Promise<ResponseEntity<ReportRecord>>} Http response containing the report record
+   */
+  @waitForValue('@')
+  restartReport(reportId) {
+    if (!reportId) {
+      throw Error('Invalid parameter(s)');
+    }
+
+    return this.request({
+      method: 'PUT',
+      service: 'ediscovery',
+      resource: `reports/${reportId}`
+    })
+      .then((res) => res)
+      .catch((reason) => {
+        this.logger.error(`Restart Report Error ${reason}`);
+        return reason;
+      });
+  },
+
+
+  /**
+   * Retrieves a list of the conversations relevant to a specified report
+   *
+   * @param {UUID} reportId - Id of the report being requested
+   * @returns {Promise<ResponseEntity<ContentSummary>>} Http response containing the content summary
+   */
+  @waitForValue('@')
+  getContentSummary(reportId) {
+    if (!reportId) {
+      throw Error('Invalid parameter(s)');
+    }
+
+    let offset = 0;
+    const size = 100;
+    let contentSummary = [];
+
+    function getNextContentSummaryPage(plugin) {
+      return plugin.request({
+        method: 'GET',
+        service: 'ediscovery',
+        resource: `reports/${reportId}/contents/summary`,
+        qs: {offset, size}
+      })
+        .then((res) => {
+          plugin.logger.info(res);
+          contentSummary = contentSummary.concat(res.body);
+          if (res.body.length < size) {
+            return contentSummary;
+          }
+          offset += size;
+          return getNextContentSummaryPage(plugin);
+        })
+        .catch((reason) => {
+          plugin.logger.error(reason);
+          return reason;
+        });
+    }
+
+    return getNextContentSummaryPage(this);
+  },
+
+  /**
+   * Retrieves a list of the conversations relevant to a specified report, paged with a specified offset and size
+   *
+   * @param {UUID} reportId - Id of the report being requested
+   * @param {long} offset - the starting index of the results the caller wants
+   * @param {long} size - the number of conversations the caller wants
+   * @returns {Promise<ResponseEntity<ContentSummary>>} Http response containing the specified content summary
+   */
+  @waitForValue('@')
+  getPagedContentSummary(reportId, offset, size) {
+    if (!reportId) {
+      throw Error('Invalid parameter(s)');
+    }
+
+    return this.request({
+      method: 'GET',
+      service: 'ediscovery',
+      resource: `reports/${reportId}/contents/summary`,
+      qs: {offset, size}
+    })
+      .then((res) => res)
+      .catch((reason) => {
+        this.logger.error(reason);
+        return reason;
+      });
+  },
+
+  /**
+   * Generates the report content and triggers the download of the content
+   *
+   * @param {UUID} reportId - Id of the report being requested
+   * @returns {List<Activities>} HttpResponse
+   */
+  @waitForValue('@')
+  generateReport(reportId) {
+    if (!reportId) {
+      throw Error('Invalid parameter(s)');
+    }
+
```

Comment:
Retrieves report content from the ediscovery service and decrypts it until the complete report is done.